site/src/sphinx/client-circuit-breaker.rst

```diff
@@ -111,40 +111,81 @@ is raised or the status is ``5xx``, succeeds when the status is ``2xx`` and igno
+                                                              @Nullable Throwable cause) {
+            if (cause != null && !(cause instanceof UnprocessedRequestException)) {
+                // A failure if an Exception is raised.
+                return CompletableFuture.completedFuture(false);
+            }
+
+            final HttpStatus status = ctx.log().responseHeaders().status();
+            if (status != null) {
+                // A failure if the response is 5xx.
+                if (status.codeClass() == HttpStatusClass.SERVER_ERROR) {
+                    return CompletableFuture.completedFuture(false);
                 }
 
-                final HttpStatus status = res.status();
-                if (status != null) {
-                    // A failure if the response is 5xx.
-                    if (status.codeClass() == HttpStatusClass.SERVER_ERROR) {
-                        return false;
-                    }
-
-                    // A success if the response is 2xx.
-                    if (status.codeClass() == HttpStatusClass.SUCCESS) {
-                        return true;
-                    }
+                // A success if the response is 2xx.
+                if (status.codeClass() == HttpStatusClass.SUCCESS) {
+                    return CompletableFuture.completedFuture(true);
                 }
+            }
 
-                // Neither a success nor a failure. Do not take this response into account.
-                return null;
-            });
+            // Neither a success nor a failure. Do not take this response into account.
+            return CompletableFuture.completedFuture(null);
         }
     };
 
 If you want to treat a :api:`Response` as a success, return ``true``. Return ``false`` to treat as a failure.
 Note that :api:`CircuitBreakerStrategy` can return ``null`` as well. It won't be counted as a success nor
 a failure.
 
+You can use the actual ``HttpResponse`` to make a decision overriding
+``needsResponseStreaming(ClientRequestContext ctx)`` to return ``true``:
+
+.. code-block:: java
+
+    import com.linecorp.armeria.common.HttpStatusClass;
+
+    final CircuitBreakerStrategy<HttpResponse> myStrategy = new CircuitBreakerStrategy<HttpResponse>() {
+
+        @Override
+        public boolean needsResponseStreaming(ClientRequestContext ctx) {
+            return true; // I want to use the actual response in the method below!
+        }
+
+        @Override
+        public CompletionStage<Boolean> shouldReportAsSuccess(ClientRequestContext ctx,
+                                                              @Nullable HttpResponse response,
+                                                              @Nullable Throwable cause) {
+            return response.aggregate().handle((res, thrown) -> {
+                if (thrown != null && !(thrown instanceof UnprocessedRequestException)) {
+                    // A failure if an Exception is raised.
+                    return false;
```

Comment:
Ditto - return null explicitly on `UnprocessedRequestException` and explain why briefly.